narwhals/_arrow/dataframe.py

```diff
@@ -387,3 +388,76 @@ def rename(self, mapping: dict[str, str]) -> Self:
     def write_parquet(self, file: Any) -> Any:
         pp = get_pyarrow_parquet()
         pp.write_table(self._native_dataframe, file)
+
+    def is_duplicated(self: Self) -> ArrowSeries:
+        from narwhals._arrow.series import ArrowSeries
+
+        np = get_numpy()
+        pa = get_pyarrow()
+        pc = get_pyarrow_compute()
+        df = self._native_dataframe
+
+        columns = self.columns
+        col_token = generate_unique_token(n_bytes=8, columns=columns)
+        row_count = (
+            df.append_column(col_token, pa.array(np.arange(len(self))))
+            .group_by(columns)
+            .aggregate([(col_token, "count")])
+        )
+        is_duplicated = pc.greater(
+            df.join(
+                row_count, keys=columns, right_keys=columns, join_type="inner"
+            ).column(f"{col_token}_count"),
+            1,
+        )
```

Comment:
wow

Reply:
Not _too_ proud of the join, but alternative was looping through value counts :|